client/src/js/services/grid/GridGrouping.js

```diff
@@ -129,31 +138,88 @@ function GridGroupingService(uiGridGroupingConstants, $filter, Session, $timeout
    */
   function configureDefaultGroupingOptions(gridApi) {
 
-    // bind the group selection method
-    gridApi.selection.on.rowSelectionChanged(null, selectAllGroupElements.bind(this));
+    //this instruction block can be executed if the grid involves selection functionality
+    if(gridApi.selection){
+
+      // bind the group selection method
+      gridApi.selection.on.rowSelectionChanged(null, selectAllGroupElements.bind(this));
+
+      gridApi.selection.on.rowSelectionChangedBatch(null, handleBatchSelection.bind(this));
+
+    }
 
     // hook into rows rendered call to ensure the grid is ready before expanding initial nodes
     gridApi.core.on.rowsRendered(null, util.once(function () {
-      gridApi.grouping.groupColumn('trans_id');
+      gridApi.grouping.groupColumn(this.column);
 
       // for the expandAllRows() to be fired last
-      $timeout(gridApi.treeBase.expandAllRows, 0, false);
-    }));
+      unfoldAll(gridApi);
+    }.bind(this)));
+  }
+
+  function unfoldAll(api) {
+    $timeout(api.treeBase.expandAllRows, 0, false);
+  }
+
+  function changeGrouping (column) {
+
+    this.gridApi.grouping.setGrouping(
+      {aggregations : [
+          {colName : 'debit_equiv', field : 'debit_equiv', aggregation : {label : 'total', type : 'sum'} },
+          {colName : 'credit_equiv', field : 'credit_equiv', aggregation : {label : 'total', type : 'sum'}}
+        ],
+
+        grouping : [
+          {colName : column, field : column, groupPriority : 0 }
+        ]
+      }
+    );
+
+    unfoldAll(this.gridApi);
+  }
+
+
+  // return the current grouping
+  function  getCurrentGroupingColumn () {
+    var groupingDetail = this.gridApi.grouping.getGrouping();
+    return groupingDetail.grouping[0].colName;
+  }
+
+  /** return back the list of selected rows **/
+
+  function getSelectedRows (){
+
+    if(!this.gridApi.selection){ return [];}
+
+    return this.gridApi.selection.getSelectedGridRows();
   }
 
   /**
    * @constructor
    */
-  function GridGrouping(gridOptions) {
+  function GridGrouping(gridOptions, isGroupHeaderSelectable, column) {
+
+    /**
+     * contains the number of selected rows
+     * TODO : create a separate service to handle selection functionnality of the grid as grouping and selection are differents
+     */
+    this.selectedRowCount = 0;
+    this.getSelectedRows = getSelectedRows.bind(this);
+    this.changeGrouping = changeGrouping.bind(this);
+    this.getCurrentGroupingColumn = getCurrentGroupingColumn.bind(this);
+    this.column = column || 'trans_id';
```

Comment:
What is `this.column`?  Is it for the default column?  Can't it be set in the `gridOptions` instead of passed in as a separate parameter?